examples/map/map_carrington_reprojection.py

```diff
@@ -0,0 +1,57 @@
+"""
+==================
+Re-projecting Maps
+==================
+
+How to reproject a map into
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+from astropy.wcs import WCS
+
+from reproject import reproject_interp
+
+import sunpy.map
+import sunpy.data.sample
+
+###############################################################################
+# We start with the sample data
+aia_map = sunpy.map.Map(sunpy.data.sample.AIA_193_IMAGE)
+
+fig = plt.figure()
+ax = plt.subplot(projection=aia_map)
+aia_map.plot(ax)
+
+###############################################################################
+# Now construct a header for the output
+shape_out = [720, 1440]
+header = sunpy.map.make_fitswcs_header(np.empty(shape_out),
```

Comment:
Just a general thing, I really don't like how we have to allocate a whole array just for this.